### ISSUE TITLE
RMap `keySet`  Codec exception

### ISSUE THREAD

**Issue by timnick-snow** [redisson/redisson#5412]
**Expected behavior**

**Actual behavior**
```text
23-11-06 11:28:13 ERROR 37464 --- [sson-netty-2-19] g.redisson.client.handler.CommandDecoder : Unable to decode data. channel: [id: 0x0c25a6f8, L:/192.168.1.18:3915 - R:redis.url/192.168.1.14:6379], reply: ReplayingDecoderByteBuf(ridx=25, widx=35), command: (EVAL), promise: java.util.concurrent.CompletableFuture@fa68e44[Not completed, 1 dependents], params: [local result = {}; local res; if (#ARGV == 3) then  res = redis.call('hscan', KEYS[1], ARGV[1], 'match', ARGV[2], 'count', ARGV[3]); else  res = redis.call('hscan', KEYS[1], ARGV[1], 'count', ARGV[2]); end;for i, value in ipairs(res[2]) do if i % 2 ~= 0 then local key = res[2][i]; table.insert(result, key); end; end;return {res[1], result};, 1, test:keySet, 0, 10]

com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'key1': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (io.netty.buffer.ByteBufInputStream); line: 1, column: 5]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1851)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:717)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._reportInvalidToken(UTF8StreamJsonParser.java:3588)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleUnexpectedValue(UTF8StreamJsonParser.java:2683)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:865)
	at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:757)
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4664)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4513)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3505)
	at org.redisson.codec.JsonJacksonCodec$2.decode(JsonJacksonCodec.java:99)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:393)
	at org.redisson.client.handler.CommandDecoder.decodeList(CommandDecoder.java:437)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:402)
	at org.redisson.client.handler.CommandDecoder.decodeList(CommandDecoder.java:437)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:402)
	at org.redisson.client.handler.CommandDecoder.decodeCommand(CommandDecoder.java:205)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:144)
```

**Steps to reproduce or test case**
```java
    @Test
    public void fun5() throws Exception {
        RedissonClient redissonClient = RedisHelp.getClient();
        JsonJacksonCodec valueCodec = new JsonJacksonCodec(new ObjectMapper());
        StringCodec keyCodec = new StringCodec();
        RMap<String, Object> map = redissonClient.getMap("test:keySet", new CompositeCodec(keyCodec, valueCodec, valueCodec));
        map.put("key1", 300L);
        map.put("key2", 400L);

        for (Object key : map.keySet()) {
            System.out.println(map.get(String.valueOf(key)));
        }
    }
```

**Redis version**
7.0.11

**Redisson version**
3.21.0

**Redisson configuration**
SingleServer

**Comment by mrniko** [comment]
unable to reproduce with the latest version

**Comment by timnick-snow** [comment]
> unable to reproduce with the latest version

Yes, I tried the latest version and everything works fine. This bug seems to have been fixed in a certain release

**Comment by mrniko** [comment]
It was fixed in 3.21.1 version https://github.com/redisson/redisson/issues/5006
